### PR TITLE
Add run SQLFlow with hive server tutorial

### LINKED ISSUE #835: Add a tutorial that introduces how to run SQLFLow with Hive via HiveServer2

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #835

### CODE REVIEW

File: doc/run_with_hive.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Run SQLFlow with Hive via HiveServer2
+
+This document is a tutorial on how to run SQLFlow, which connects to the hive server2.
+
+For the most production environment, the system administrators may setup hive server with [authentication configuration](https://cwiki.apache.org/confluence/display/Hive/Setting+Up+HiveServer2#SettingUpHiveServer2-Authentication/SecurityConfiguration): e.g., KERBEROS, LDAP, PAM, or CUSTOM.
+
+## Connect Hive Server with No SASL
+
+Launch your standalone hive server Docker container by running:
```
Comment (typhoonzero) [inline doc/run_with_hive.md:39]: Maybe add some explanation that this is for test only, user can connect to a production hive server according to their own setup.

File: doc/run_with_hive.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+``` bash
+> docker run -d -p 8888:8888 --name=hive sqlflow/gohive:dev python3 -m http.server 8899
```
Comment (tonyyang-svail) [inline doc/run_with_hive.md]: Can we remove `python3 -m http.server 8899`? I think the HTTP server is used only in our test script.

https://github.com/sql-machine-learning/sqlflow/blob/0efd29902a371ce7cc1fe9e6ce9f9c1de5395472/scripts/test_hive.sh#L15-L26

File: doc/run_with_hive.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Run SQLFlow with Hive via HiveServer2
+
+This document is a tutorial on how to run SQLFlow, which connects to the hive server2.
+
+For the most production environment, the system administrators may setup hive server with [authentication configuration](https://cwiki.apache.org/confluence/display/Hive/Setting+Up+HiveServer2#SettingUpHiveServer2-Authentication/SecurityConfiguration): e.g., KERBEROS, LDAP, PAM, or CUSTOM.
```
Comment (tonyyang-svail) [inline doc/run_with_hive.md]: Have we supported KERBEROS, LDAP, ...? If we have supported them all, we should provide specific examples.

**Review by tonyyang-svail — COMMENTED** [review]
I believe the reader of this document is the user of SQLFlow, so the document should focus on deploying SQLFlow with their **existing** Hive service. While the content looks more like "Hi SQLFlow developers, here are two examples of setting up SQLFlow - Hive testing environment". 

We should assume the reader is not familiar with SQLFlow, so every new concept needs a detailed explanation. For example, when we are mentioning `--datasource='hive://root:root@localhost:10000/'`, we should explain the data source is in the format of `hive://user:password@address/database?param_n=arg_n`, so that the user can adjust according to his/her own use case.

File: doc/run_with_hive.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+This implies settings in `hive-site.xml`:
+
+``` text
+hive.server2.authentication = NOSASL
+```
+
+Test SQLFlow by running a query in Jupyter Notebook
+
+``` bash
+> docker run --rm --net=container:hive sqlflow/sqlflow \
```
Comment (tonyyang-svail) [inline doc/run_with_hive.md:54]: Do we need to add `-p 8888:8888` so that the Jupyter server in the container can be accessed by the browser on the host?

**Comment by tonyyang-svail** [comment]
@Yancey1989 Maybe the following structure would help.

--------------------------------------------

# Run SQLFlow with Hive via HiveServer2

This tutorial explains how to connect SQLFlow with Hive via HiveServer2. It has been tested on the Hive version [...].

## Connect existing Hive Service

To connect an existing Hive Service, we only need to configure a data source string in the format of

```
hive://user:password@address/dbname[?param1=value1&...&paramN=valueN]
```

The data source string contains the credential and the configurations for connecting Hive. For example, if we want to connect the database `iris` at `127.0.0.1:10000` using `root` as username and password, we can write `hive://root:root@127.0.0.1:10000/iris` as the data source string. You can find more configurations options at [gohive](https://github.com/sql-machine-learning/gohive).

Using the data source string, we can start an all-in-one SQLFlow container by running

```bash
docker run --rm -p 8888:8888 sqlflow/sqlflow bash -c \
"sqlflowserver --datasource='hive://root:root@localhost:10000/iris' &
SQLFLOW_SERVER=localhost:50051 jupyter notebook --ip=0.0.0.0 --port=8888 --allow-root --NotebookApp.token=''"
```

Then we can open a web browser and go to `localhost:8888`. There are many SQLFlow tutorials, e.g. `tutorial_dnn_iris.ipynb`. We can follow the tutorials and substitute the data for our own use.

## Connect standalone Hive server for testing

We also pack a Hive server in a Docker image for testing. ....


--------------------------------------------

@typhoonzero @weiguoz Maybe we should let MySQL and MaxCompute tutorials follow a similar structure?

File: doc/run_with_hive.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# Run SQLFlow with Hive via HiveServer2
```
Comment (Yancey0623) [inline doc/run_with_hive.md]: How to connect Hive with SQLFlow

File: doc/run_with_hive.md (hunk 2 above)
Comment (Yancey0623) [reply doc/run_with_hive.md]: Sure, will do that.

File: doc/run_with_hive.md (hunk 3 above)
Comment (Yancey0623) [reply doc/run_with_hive.md]: `sql-machine-learning/gohive` use [beltran/gohive](https://github.com/beltran/gohive#supported-connections) to connect hiveserver2 with SASLTransport, I think we need to add more test in `sql-machine-learning/gohive` to make sure it works well in SQLFlow, before that we can leave one example (PAM auth) here.

File: doc/run_with_hive.md (hunk 4 above)
Comment (Yancey0623) [reply doc/run_with_hive.md:54]: The sqlflow Docker container shared the network stack of hive container by `--net=container:hive ` and the hive container exposed the port to host `-p  8888:8888`.

**Comment by Yancey0623** [comment]
Thanks @tonyyang-svail , I updated this PR followed your comment.

File: doc/run_with_hive.md (hunk 1 above)
Comment (Yancey0623) [reply doc/run_with_hive.md:39]: done.

File: doc/run_with_hive.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+Using the `datasource` string, we can launch an all-in-one Docker container by running:
+
+``` bash
+docker run --rm -p 8888:8888 sqlflow/sqlflow bash -c \
+"sqlflowserver --datasource='hive://root:root@localhost:10000/iris' &
+SQLFLOW_SERVER=localhost:50051 jupyter notebook --ip=0.0.0.0 --port=8888 --allow-root --NotebookApp.token=''"
+```
+
+Then we can open a web browser and go to `localhost:8888`. There are many SQLFlow tutorials, e.g. `tutorial_dnn_iris.ipynb`. We can follow the tutorials and substitute the data for our own use.
```
Comment (typhoonzero) [inline doc/run_with_hive.md]: Then we can => Then you can ...

Comment (Yancey0623) [reply doc/run_with_hive.md]: done.

File: doc/run_with_hive.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+# How SQLFlow connects with Hive
+
+This document is a tutorial on how SQLFlow connects Hive via [HiveServer2](https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Overview).
+
+## Connect Existing Hive server
+
+To connect an existing Hive server instance, we only need to configure a `datasource` string in the format of
+
+``` text
+hive://user:password@ip:port/dbname[?auth=<auth_mechanism>&session.<cfg_key1>=<cfg_value1>...&session<cfg_keyN>=valueN]
```
Comment (tonyyang-svail) [inline doc/run_with_hive.md:10]: Missing dot: `session<cfg_keyN>=valueN]`=>`session.<cfg_keyN>=valueN]`

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

**Review by typhoonzero — APPROVED** [review]

**Review by weiguoz — APPROVED** [review]

**Pull request merged**